.github/workflows/continuous-integration.yml

```diff
@@ -68,24 +68,21 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
+      - name: Base Setup
+        uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
         with:
-          python-version: ${{ matrix.python-version }}
+          python_version: ${{ matrix.python-version }}
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          # All dependencies but markdown-it-py
-          python -m pip install nbformat pyyaml toml
-          python -m pip install -r requirements-dev.txt
           # install sphinx_gallery and matplotlib if available
           python -m pip install sphinx_gallery~=0.7.0 || true
           python -m pip install jupyter-fs || true
       - name: Install markdown-it-py
         if: ${{ matrix.markdown-it-py-version }}
         run: python -m pip install markdown-it-py${{ matrix.markdown-it-py-version }}
       - name: Install from source (required for the pre-commit tests)
-        run: python -m pip install . --no-deps
+        run: python -m pip install '.[dev]'
```

Comment:
Not sure where it is relevant, but where you need to test both `labextension` and the main python project use something like
```console
$ pyhon -m pip install -e .[test] -e ./jupyterlab-jupytext[test] ...
```